src/fetch.js

```diff
@@ -26,7 +26,12 @@ module.exports = async ({ apiURL, contentType, singleType, jwtToken, queryLimit,
  */
 const clean = item => {
   forEach(item, (value, key) => {
-    if (startsWith(key, `__`)) {
+    // Remove mongo's __v
+    if (key === `__v`) {
+      delete item[key]
+    // Output type for strapi's dynamic components
+    } else if (key === `__component`) {
+      item.dynamicComponentType = item[key]
```

Comment:
Just wondering but an't we just leave the __component field ?

Reply:
Hmm the following if clause makes me think we're trying to stear clear of underscore-prepended fields, but I think that should be your call :)

Reply:
Yeah tbh I would clear this as I don't see the point of all this... I'm just wondering if this was done to avoid bugs with gatsby (or graphql) not liking underscores as first character... Would you mind trying it out ?

Reply:
Just tried it and yeah, it will output, but is of course converted to `_component` due to the following if clause. I feel that's a pretty weird side-effect.

This https://github.com/strapi/gatsby-source-strapi/commit/3ac3d8b171f3fc32f3c3d277b0b428132f2d7f8d is all the context I could find on this logic. I guess this is also mongoose related, and should be handled with more specificity I think. See new commit.

@pierreburgy , is this something you can chime in on? Thanks :)

Reply:
Woah, now that it's not removing the first underscore, it seems that Gatsby renames `__component` to `_xcomponent`. Haven't dug too deep into this, but it seems that Gatsby reserves `__`. Also a pretty weird side-effect. I'd say going with a more specifically named key like my initial commit would actually be appropriate in this case.

Reply:
OK maybe the __ keys other than __v could be prefixed with strapi so we know they are specific to strapi

Reply:
Good call. Check my newest commit.